File: challenge_3.py
#!/usr/bin/python

import sys

def print_comb(input_str, start):
    """Returns the combination of the string for the characters X"""
    if start == len(input_str):
        print ''.join(input_str)
        return
    if input_str[start] == 'X':
        input_str[start] = '0'
        print_comb(input_str, start + 1)

        input_str[start] = '1'
        print_comb(input_str, start + 1)
        input_str[start] = 'X'
    else:
        print_comb(input_str, start + 1)


if len(sys.argv) == 2:
    print_comb(list(sys.argv[1]), 0)

File: challenge_3.js
#!/usr/bin/node

function print_comb(input, start) {        
    if(start === input.length) {
        console.log(input.join(''));
        return;
    }

    if(input[start] === 'X') {
       input[start] = '1';
       print_comb(input, start + 1);

       input[start] = '0';
       print_comb(input, start + 1);
       input[start] = 'X'
    } else {
        print_comb(input, start + 1);
    }
}

if(process.argv.length === 3) {
    print_comb(Array.from(process.argv[2]), 0)
}
